src/vs/workbench/contrib/terminal/browser/xterm/quickFixAddon.ts

```diff
@@ -130,26 +108,16 @@ export class TerminalQuickFixAddon extends Disposable implements ITerminalAddon,
 			return;
 		}
 		this._register(commandDetection.onCommandFinished(command => {
-			if (this._fixesAvailable) {
-				if (this._fixesShown) {
-					if (this._expectedCommands?.includes(command.command)) {
-						this._logService.debug(QuickFixCommandResultTelemetryEvent.QuickFixExecutedManually);
-						this._telemetryService?.publicLog2<{ classification: 'SystemMetaData'; purpose: 'FeatureInsight' }>(QuickFixCommandResultTelemetryEvent.QuickFixExecutedManually);
-					} else {
-						this._logService.debug(QuickFixCommandResultTelemetryEvent.QuickFixWrong);
-						this._telemetryService?.publicLog2<{ classification: 'SystemMetaData'; purpose: 'FeatureInsight' }>(QuickFixCommandResultTelemetryEvent.QuickFixWrong);
-					}
-					this._fixesShown = false;
-				} else {
-					this._logService.debug(QuickFixCommandResultTelemetryEvent.QuickFixIgnored);
-					this._telemetryService?.publicLog2<{ classification: 'SystemMetaData'; purpose: 'FeatureInsight' }>(QuickFixCommandResultTelemetryEvent.QuickFixIgnored);
-				}
-			} else {
+			if (this._expectedCommands) {
+				const info = `terminal/quick-fix/command/fixes-shown:${this._fixesShown}/expected-command:${this._expectedCommands.includes(command.command)}`;
+				this._logService.info(info);
+				this._telemetryService?.publicLog2<{ classification: 'SystemMetaData'; purpose: 'FeatureInsight' }>(info);
```

Comment:
We still want the events to be called something like `terminal/quick-fix/command/fixes-shown`, but you can add additional metadata to the event which is filterable. Here's an example:

https://github.com/microsoft/vscode/blob/c5aa167e9b1c260f2b26b5fb3d93683ca5484b7e/src/vs/code/electron-main/app.ts#L1211-L1217